File: lab4/Figures/Domain/Cone/Cone.cpp
#include "Cone.hpp"
#include "../../../../Utils/StringUtils.hpp"

double Cone::GetHeight() const
{
    return height;
}

Cone::operator std::string() const
{
    return StringUtils::StringConcatenator() << density <<  ' ' << volume << ' ' << GetMass() << ' ' << height;
}

File: lab4/Bodies/lib/Figures/Cone/Cone.dart
import '../BodyRotation/BodyRotation.dart';

class Cone extends BodyRotation
{
  double _height;

  Cone(double density, double volume, double radius, double this._height): super(density, volume, radius) {}

  double GetHeight() => _height;

  @override
  String toString() {
    return 'Cone: ' + 'V = ' + GetVolume().toString() + ' R = ' + GetBaseRadius().toString() +
    ' Dens = ' + GetDensity().toString() + ' H = ' + _height.toString();
  }
}
